pymeasure/adapters/ni_gpib_232ct.py

```diff
@@ -0,0 +1,346 @@
+        """
+        self.connection.write(f"trg {self.address}")
+
+    def clear(self):
+        """
+        Clear specified device.
+
+        """
+        self.connection.write(f"clr {self.address}")
+
+    @property
+    def eoi(self):
+        """Control whether to assert the EOI signal with the last character
+        of any command sent over GPIB port (bool).
+
+        Some instruments require EOI signal to be
+        asserted in order to properly detect the end of a command.
+        """
+        self.connection.write("EOT")
+        return bool(int(self.connection.read()))
+
+    @eoi.setter
+    def eoi(self, value):
+        self.connection.write(f"EOT {int(value)}")
+
+    def gpib(self, address, **kwargs):
+        """Return a NI_GPIB_232 object that references the GPIB
+        address specified, while sharing the Serial connection with other
+        calls of this function.
+
+        :param address: Integer GPIB address of the desired instrument
+        :param kwargs: Arguments for the initialization
+        :returns: NI_GPIB_232 for specific GPIB address
+        """
+        return NI_GPIB_232(self, address, **kwargs)
+
+    def pass_control(self, primary_address: int, secondary_address: int):
+        """
+        Pass control to device with primary_address and optional secondary_address.
+
+        """
+        self.connection.write(f"pct {primary_address}+{secondary_address}")
+
+    def send_command(self, data: bytes):
+        """
+        Write GPIB command bytes on the bus.
+
+        """
+        self.connection.write(f"cmd #{len(data)}\n {data}")
+        self._check_errors()
+
+    def set_rsc(self):
+        """
+        Set the NI-GPIB232ct to become the GPIB system controller.
+
+        """
+        self.connection.write("rsc 1")
+
+    def send_ifc(self):
+        """Pulse the interface clear line (IFC) for at least 200 microseconds."""
+        self.connection.write("sic 0.0002")
+
+    @property
+    def timeout(self):
+        """Control the GPIB timeout.
+        Valid value range 0.0001 to 3600s
+        """
+        self.connection.write("tmo")
+        ret_val = self.connection.read().split(",")
+        return float(ret_val[0])
+
+    @timeout.setter
+    def timeout(self, value):
+        value = strict_range(value, [0.0001, 3600])
+        if value < 1:
+            self.connection.write(f"tmo {value:.4f}")
+        elif value <= 3600:
+            self.connection.write(f"tmo {int(value)}")
+
+    @property
+    def version(self):
+        """Get the version string of the NI GPIB-232-CT."""
+        self.flush_read_buffer()
+        self.connection.write("id \r")
+        time.sleep(0.02)
+        return self.connection.read_bytes(71).decode()
+
+    def wait_for_srq(self, timeout=20, delay=0.1):
+        """Blocks until a SRQ, and leaves the bit high
+
+        :param timeout: Timeout duration in seconds.
+        :raises TimeoutError: "Waiting for SRQ timed out."
```

Comment:
What's the meaning of the return value (`None` or integer)?

Reply:
- pylint was complaining without the `return None`
- it could happen that no SRQ signal is caught (case None) or the status byte is returned

Reply:
OK; you could document that in the docstring.

Reply:
From what I see in the logic of the method, a `ret_val` of `0` as received from the hardware is not possible or invalid.
You could return `0` instead of `None` if no signal is caught, then the return type remains homogeneous (which makes it easier to deal with downstream). Does that make sense in the context of SRQ values?

Reply:
Another tack: what do you think about using the same interface/behaviour as https://pyvisa.readthedocs.io/en/latest/api/resources.html#pyvisa.resources.GPIBInstrument.wait_for_srq ? It does the same thing afaict. Do you need the additional code flow handling (exceptions, return None, getting the received value, etc)?